lib/AppInfo/Application.php

```diff
@@ -176,7 +179,24 @@ public function __construct (array $urlParams = array()) {
 			$server = $c->query('ServerContainer');
 
 			$user = $server->getUserSession()->getUser();
-			return ($user) ? $user->getUID() : '';
+			if ($user instanceof IUser) {
+				return (string) $user->getUID();
+			}
+
+			$request = $server->getRequest();
+			if (!empty($request->server['PHP_AUTH_USER'])) {
+				$token = $request->server['PHP_AUTH_USER'];
+				try {
+					$share = $server->getShareManager()->getShareByToken($token);
+					if ($share->getShareType() === Share::SHARE_TYPE_REMOTE) {
```

Comment:
Yep that is good. We use those constants all over the place. I might move them to the share manager at some point. But we need a deprecation period anyways.